src/pkg/services/m365/api/users.go

```diff
@@ -41,8 +41,6 @@ type Users struct {
 
 type UserInfo struct {
 	DiscoveredServices map[path.ServiceType]struct{}
-	HasMailBox         bool
-	HasOneDrive        bool
```

Comment:
Since DiscoveredServices is already having services info, it makes a lot of sense to not have these flags. 
But aren't we adding a extra burden on functions like `checkServiceEnabled` which used to call the `UserInfo` function for just getting `DiscoveredServices` info and are not fetching the MailBoxSettings too?

Infact `checkServiceEnabled` will call the `UserInfo` function to know about one service at a time(this is my understanding of the code and could be wrong) why burden it with other calls?